src/repository/logic.py

```diff
@@ -367,12 +370,13 @@ def get_published_preprints(request, user_subject_pks):
         return published_preprints.filter(pk__in=user_subject_pks)
 
 
-def get_preprint_if_id(preprint_id):
+def get_preprint_if_id(preprint_id, repository):
```

Comment:
Is this function even eeded? sounds like the one view using it could do the `if preprint_id` check itself

Reply:
Dropped. Thought it would be used more, was wrong XD